5.5/docker-entrypoint.sh

```diff
@@ -38,25 +24,25 @@ if [ "$1" = 'mysqld' ]; then
 		mysql_install_db --user=mysql --datadir="$DATADIR" --rpm --basedir=/usr/local/mysql
 		echo 'Finished mysql_install_db'
 
-		mysqld --user=mysql --datadir="$DATADIR" --skip-networking --basedir=/usr/local/mysql --pid-file="$PIDFILE" &
-		for i in $(seq 30 -1 0); do
-			[ -S "$SOCKET" ] && break
+		mysqld --user=mysql --datadir="$DATADIR" --skip-networking --basedir=/usr/local/mysql &
+		pid="$!"
```

Comment:
By using `$!` instead of parsing the `mysqld` config, we can guarantee that the process is a child of our shell and thus we can `wait` on it during shutdown (see below) instead of looping while we wait for it to die.